source/retryable-reads/tests/unified/listCollectionObjects.yml

```diff
@@ -0,0 +1,136 @@
+# listCollectionObjects returns an array of MongoCollection objects.
+# Not all drivers support this functionality. For more details, see:
+# https://github.com/mongodb/specifications/blob/v1/source/enumerate-collections.rst#returning-a-list-of-collection-objects
+
+description: listCollectionObjects
+
+schemaVersion: '1.9'
+
+runOnRequirements:
+  -
+    minServerVersion: '4.0'
+    topologies:
+      - single
+      - replicaset
+  -
+    minServerVersion: 4.1.7
+    topologies:
+      - sharded
+      - load-balanced
+
+createEntities:
+  -
+    client:
+      id: &client0 client0
+      useMultipleMongoses: false
+      observeEvents:
+        - commandStartedEvent
+  -
+    database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &database_name retryable-reads-tests
+
+initialData:
+  -
+    collectionName: &collection_name coll
+    databaseName: *database_name
+    documents: []
+
+tests:
+  -
+    description: 'ListCollectionObjects succeeds on first attempt'
+    operations:
+      - &retryable_operation
+        object: *database0
+        name: listCollectionObjects
```

Comment:
We already have `listCollections` in other spec tests.  Is `listCollectionObjects` just an alias for that, and if so, is it necessary?

Same for `listDatabaseObjects`.

Reply:
I see.  If a driver doesn't offer those methods, I guess it can skip the tests of them, as they appear to be the same set of tests as for the non-object variants.

Reply:
> If a driver doesn't offer those methods, I guess it can skip the tests of them

Yes. AFAIK, that is the standard practice for any driver when dealing with tests for unsupported/unimplemented functionality.